src/pages/LoginPage/LoginPage.jsx

```diff
@@ -112,7 +115,15 @@ const LoginPage = ({ isFirstTime }) => {
   const handleSubmit = (e) => {
     e.preventDefault()
 
-    doLogin()
+    if (!usernameInput?.current?.value) {
```

Comment:
Why are you using refs here? We already know the name and password from the state. This should be a one line `if` fix.

Reply:
I am using refs for the focus part, so if you click on login button and username is empty, it will focus you to the input box :)

Reply:
Oh I see, sorry missed the focus line. I think that's overkill imo. There are only two fields, the user can just click the empty one. So please remove the refs to clean up the code and use the state to check the fields.

Reply:
Made the changes, Thank you for your suggestions, It reduced the lines of code by a good margin.